src/compiler/checker.ts

```diff
@@ -16298,7 +16298,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function getTypeArguments(type: TypeReference): readonly Type[] {
         if (!type.resolvedTypeArguments) {
             if (!pushTypeResolution(type, TypeSystemPropertyName.ResolvedTypeArguments)) {
-                return type.target.localTypeParameters?.map(() => errorType) || emptyArray;
+                return concatenate(type.target.outerTypeParameters, type.target.localTypeParameters?.map(() => errorType) || emptyArray);

```

Comment:
No need for the `emptyArray` bit given the parameter accepts undefined.

Reply:
it's actually kinda needed so `concatenate` doesn't return `undefined`. If I remove it get type errors here

Reply:
Could stick it on the outside, maybe...

Reply:
I'm not sure what do you mean

Reply:
As in `return concatenate(a, b) ?? emptyArray`

But maybe that's worse. Can't recall if this helper will make a spurious copy on an empty input array or not